pkg/controller/seed-controller-manager/backup/backup_controller.go

```diff
@@ -285,12 +282,8 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, clus
 	}
 
 	// Always add the finalizer first
-	if !kuberneteshelper.HasFinalizer(cluster, cleanupFinalizer) && !r.disabled {
-		oldCluster := cluster.DeepCopy()
-		kuberneteshelper.AddFinalizer(cluster, cleanupFinalizer)
-		if err := r.Patch(ctx, cluster, ctrlruntimeclient.MergeFrom(oldCluster)); err != nil {
-			return fmt.Errorf("failed to update cluster after adding cleanup finalizer: %w", err)
-		}
+	if err := kuberneteshelper.TryAddFinalizer(ctx, r, cluster, cleanupFinalizer); err != nil {
+		return fmt.Errorf("failed to add finalizer: %w", err)
```

Comment:
This change lost the `&& !r.disabled` condition. Is that okay?

Reply:
Oh wow, well spotted!

Reply:
Fixed.